tokio/src/task/core.rs

```diff
@@ -55,102 +54,213 @@ pub(crate) struct Header {
 
     /// Table of function pointers for executing actions on the task.
     pub(super) vtable: &'static Vtable,
-
-    /// Used by loom to track the causality of the future. Without loom, this is
-    /// unit.
-    pub(super) future_causality: CausalCell<()>,
 }
 
 /// Cold data is stored after the future.
 pub(super) struct Trailer {
     /// Consumer task waiting on completion of this task.
-    pub(super) waker: CausalCell<MaybeUninit<Option<Waker>>>,
+    pub(super) waker: CausalCell<Option<Waker>>,
 }
 
 /// Either the future or the output.
-enum Stage<T: Future> {
-    Running(Track<T>),
-    Finished(Track<super::Result<T::Output>>),
+pub(super) enum Stage<T: Future> {
+    Running(T),
+    Finished(super::Result<T::Output>),
     Consumed,
 }
 
-impl<T: Future> Cell<T> {
+impl<T: Future, S: Schedule> Cell<T, S> {
```

Comment:
Any particular reason why this `impl` requires `S: Schedule`?

Reply:
`vtable` needs it now. This is because the cell contains a `S` and doesn't erase it as a `*const ()` like it did before.